api/src/Entity/ContentNode.php

```diff
@@ -56,29 +53,17 @@ abstract class ContentNode extends BaseEntity implements BelongsToCampInterface,
     #[ApiProperty(writable: false, example: '/content_nodes/1a2b3c4d')]
     #[Gedmo\SortableGroup] // this is needed to avoid that all root nodes are in the same sort group (parent:null, slot: '')
     #[Groups(['read'])]
-    #[ORM\ManyToOne(targetEntity: ContentNode::class, inversedBy: 'rootDescendants')]
+    #[ORM\ManyToOne(targetEntity: ColumnLayout::class, inversedBy: 'rootDescendants')]
     #[ORM\JoinColumn(nullable: true)] // TODO make not null in the DB using a migration, and get fixtures to run
-    public ContentNode $root;
-
-    /**
-     * All content nodes that are part of this content node tree.
-     */
-    #[ApiProperty(readable: false, writable: false)]
-    #[ORM\OneToMany(targetEntity: ContentNode::class, mappedBy: 'root')]
-    public Collection $rootDescendants;
+    public ?ColumnLayout $root = null;
 
     /**
      * The parent to which this content node belongs. Is null in case this content node is the
      * root of a content node tree. For non-root content nodes, the parent can be changed, as long
-     * as the new parent is in the same camp as the old one. A content node is defined as root when
-     * it has an owner.
+     * as the new parent is in the same camp as the old one.
      */
-    #[AssertEitherIsNull(
-        other: 'owner',
-        messageBothNull: 'Must not be null on non-root content nodes.',
-        messageNoneNull: 'Must be null on root content nodes.'
-    )]
-    #[AssertBelongsToSameOwner(groups: ['update'])]
+    #[Assert\NotNull(groups: ['create'])] // Root nodes have parent:null, but manually creating root nodes is not allowed
+    #[AssertBelongsToSameRoot(groups: ['update'])]
```

Comment:
Alternative name: AssertNoRootChange
// cannot change non-null value to null value